src/main/java/automatabuilder/DFA.java

```diff
@@ -15,20 +16,44 @@
 
     protected IShowDFA outputter;
 
+    public DFA(List<IState> states, IAlphabet alphabet, IState startState, IShowDFA outputter) {
+        this.states = states;
+        this.alphabet = alphabet;
+        this.startState = startState;
+        this.outputter = outputter;
+    }
+
+    public DFA(List<IState> states, IAlphabet alphabet, IState startState) {
+        this(states, alphabet, startState, null);
+    }
+
+    
+    
     protected IState delta(IState q, Symbol s) {
-        //TODO: Implement
-        return null;
+        return q.transition(s);
     }
 
     protected IState deltaHat(IState q, IWord ax) {
-        //TODO: Implement
-        return null;
+        if (ax == null || ax.equals(Word.Empty)) {
```

Comment:
Most of the time, yes. As long as we use Word.Empty to denote the empty word (which we do for the last bit when creating a word). But we should override equals just in case. I'll make an issue about it